### PR TITLE
[UnderlineNav]: VRT Snapshot and interaction Tests (Playwright)

### PR DESCRIPTION
**Pull request by broccolinisoup**
This PR is a follow up on @joshblack's [playwright PR](https://github.com/primer/react/pull/2520)
 and covers UnderlineNav related next steps.

- Introduce primer layout primitives
- Tests for states of UnderlineNav on available Primer layout breakpoints and themes (5 breakpoint and 9 themes)
- Interaction tests 

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: e2e/components/UnderlineNav.test.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import {test, expect} from '@playwright/test'
+import {visit} from '../test-helpers/storybook'
+import {themes} from '../test-helpers/themes'
+import {viewports} from '../test-helpers/viewports'
+
+test.describe('UnderlineNav', () => {
+  Object.keys(viewports).forEach(key => {
```
Comment (joshblack) [inline e2e/components/UnderlineNav.test.ts:7]: If it helps, another option could be:

```suggestion
  Object.entries(viewports).forEach(([viewport, size])=> {
```

File: e2e/components/UnderlineNav.test.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    themes.forEach(theme => {
+      test(`VRT @ ${key} (${viewports[key]}px) on ${theme} theme`, async ({page}) => {
```
Comment (joshblack) [inline e2e/components/UnderlineNav.test.ts:9]: What do you think of having the test structure be something like:

- Top-level: component name
- First-level: story name
- Second-level: theme + modifiers + tags

For example:

```ts
test.describe('UnderlineNav', () => {
  test.describe('Internal Responsive Nav', () => {
    themes.forEach(theme => {
      Object.entries(viewports).forEach(([viewport, width]) => {
        test(`${theme} (${viewport}) @vrt`, async ({ page }) => {
        })
      })
    });
  });
});
```

File: e2e/components/UnderlineNav.test.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  test('Hide Icons when there is not enough space to display all list items', async ({page}) => {
+    await visit(page, {
+      id: 'components-underlinenav--internal-responsive-nav'
+    })
+
+    // Default State
+    expect(await page.screenshot()).toMatchSnapshot()
+
+    // Resize
+    await page.setViewportSize({width: viewports['primer.breakpoint.md'], height: 768})
```
Comment (joshblack) [inline e2e/components/UnderlineNav.test.ts:52]: Would it be helpful to run axe at all during things like resize or when the menu is open, or not really?

**Review by joshblack — COMMENTED** [review]
So cool! Just left a couple comments/questions, can't wait to hear what you think 👀

File: e2e/components/UnderlineNav.test.ts (hunk 1 above)
Comment (broccolinisoup) [reply e2e/components/UnderlineNav.test.ts:7]: Nice ✨

File: e2e/components/UnderlineNav.test.ts (hunk 2 above)
Comment (broccolinisoup) [reply e2e/components/UnderlineNav.test.ts:9]: Ahh I guess I didn't think that we need to cover every story to VRT test 🙈  I rather chose one story to test the themes and viewports and write possible interaction tests based on the same story but I see now that is not the best approach. Thanks for mentioning that and yes your suggested structure makes perfect sense. I'll push a commit to update them

**Comment by broccolinisoup** [comment]
Closing this because we have the e2e interaction test in this [PR](https://github.com/primer/react/pull/2632)

**Pull request closed without merging**